src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/AbstractRatingCalculator.java

```diff
@@ -0,0 +1,77 @@
+package com.sap.sgs.phosphor.fosstars.tool.github;
+
+import com.sap.sgs.phosphor.fosstars.data.NoUserCallback;
+import com.sap.sgs.phosphor.fosstars.data.UserCallback;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import org.kohsuke.github.GitHub;
+
+/**
+ * A base class for rating calculators.
+ */
+abstract class AbstractRatingCalculator {
+
+  /**
+   * An interface to GitHub.
+   */
+  final GitHub github;
+
+  /**
+   * A token for accessing the GitHub APIs.
+   */
+  String token;
```

Comment:
We don't need any scope for this token variable?

Reply:
The field is used by classes which extend `AbstractRatingCalculator`. Technically, the field could have `protected` score. However, the classes which extend `AbstractRatingCalculator` are in the same package, so the default scope can be used as well. Since the default scope is more restrictive than `protected`, I would prefer to use the default scope.

Reply:
Ok, thanks for the explanation. 
> However, the classes which extend AbstractRatingCalculator are in the same package,
Hmm no plans of allowing the abstract class to be extended in other packages in future?

Reply:
No plans at the moment. Right now it's not public API, the `AbstractRatingCalculator` class is just used to store some common code.